src/main/java/com/amazon/ionpathextraction/SearchPath.java

```diff
@@ -46,6 +48,19 @@ int size() {
         return pathComponents.size();
     }
 
+    /**
+     * Produces a "normalized" path for the SearchPath.
+     * Basically: the SearchPath has the annotations (or not) for matching top-level-values.
+     * The "normalized" path treats this as an explicit Wildcard step and adds it to the head
+     * of the PathComponents.
+     */
+    List<PathComponent> getNormalizedPath() {
```

Comment:
What is the practical reason why this is needed by the FsmPathExtractor but not the PathExtractorImpl?

Reply:
The `PathExtractorImpl` uses the `SearchPath` to hold and manage the Annotations matching for TopLevelValues. The `FsmPathExtractor` treats matching Top-Level-Values, annotations or not, as just another `PathComponent`; it is a `Wildcard` or `Annotations` match, the same as any other nested step. I think it's a simplifier, but it does take a small shift in thinking.

Reply:
But I realize that's not a reason why it's needed. It just makes the code cleaner. Looking at it, I can just manage this in the `FsmMatcherBuilder` and avoid exposing more to this internal but public API.

Reply:
Well I did try to avoid this, but I practically needed to expose the `Annotions` and `PathComponents` from the `SearchPath` anyway. And given my initial approach was cleaner it seemed preferable.

Which brings me to the implicit insight in my initial reply: in the `PathExtractorImpl` the `SearchPath`s _do_ the filtering. In this implementation, there are a DTO to represent the result of parsing the users' expressions. I needed to expose some more data to do that and this just seemed like a pragmatic choice. 🤷